### DISCUSSION TITLE
Custom plugin - Debug with Delve

### DISCUSSION THREAD

**Discussion by fdebuire** [terraform-linters/tflint#2113] [Q&A]
Hello,

I am writing a custom plugin for tflint and have issues debugging it. `TFLINT_LOG=debug` has its limits.

Anyone knows how to debug the plugin using Delve? I am mostly using Intellij/Goland but can use VsCode if needed.

Thanks

**Comment by wata727** [comment]
Unfortunately, I spend most of my time doing log-based debugging. There may be a way to debug using a debugger even as it is now, but this was not taken into consideration when the SDK was designed.

The Terraform provider has a debugger-based debugging way, so we may be able to use the debugger by introducing a similar mechanism.
https://developer.hashicorp.com/terraform/plugin/debugging#debugger-based-debugging
